### ISSUE TITLE
Interpolating endpoint path parameters in @RolesAllowed

### ISSUE THREAD

**Issue by fcamblor** [restx/restx#216]
In some cases, it would be interesting to integrate in user roles some resource ids, and be capable to check these "dynamic" roles with `@RolesAllowed` annotation

I imagine something like this 

```
@RolesAllowed({"CAN_READ_FOO_{id}"})
@GET("/foo/{id}")
public Foo fetchFooById(String id){
    ...
}
```

In a first implementation, we could imagine providing only path parameter to the interpolation context which would (I think) represent 95% of the usages.

In a later implementation, we could imagine to interpolate query & body parameters (I imagine this should be considered as an edge case as, today, we don't consider query & body parameters attributes in the url)

WDYT ?

**Comment by a-peyrard** [comment]
Seems nice to me, would enjoy to have a feature like this.

And I agree for a two steps feature, first query params, and in a second phase query & body parameters.

**Comment by fcamblor** [comment]
Fixed through #223 PR
